clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupWithArtifactsSpec.java

```diff
@@ -367,4 +367,131 @@ public void createServerGroup_errorIfNoContainersTest() throws IOException, Inte
         String.format("Failed to detect task failure, in %s seconds", TASK_RETRY_SECONDS),
         TASK_RETRY_SECONDS);
   }
+
+  @DisplayName(
+      ".\n===\n"
+          + "Given description w/ task def artifacts, EC2 launch type, and target group mappings, "
+          + "with multiple load balancers and multiple containers successfully submit createServerGroup operation"
+          + "\n===")
+  @Test
+  public void createServerGroup_ArtifactsEC2TgMappingsWithMultipleLBsAndContainersTest()
+      throws IOException, InterruptedException {
+
+    // given
+    String url = getTestUrl(CREATE_SG_TEST_PATH);
+    String requestBody =
+        generateStringFromTestFile(
+            "/createServerGroup-artifact-EC2-TGMappings-multipleLBsAndContainers.json");
+    String expectedServerGroupName =
+        "ecs-integArtifactsEC2TgMappingsStackWithMultipleLBsAndContainers-detailTest-v000";
+
+    ByteArrayInputStream byteArrayInputStreamOfArtifactsForEC2Type =
+        new ByteArrayInputStream(
+            generateStringFromTestArtifactFile(
+                    "/createServerGroup-artifact-EC2-TGMappings-multipleLBsAndContainers-artifactFile.json")
+                .getBytes());
+
+    when(mockArtifactDownloader.download(any(Artifact.class)))
+        .thenReturn(byteArrayInputStreamOfArtifactsForEC2Type);
+
+    String taskId =
+        given()
+            .contentType(ContentType.JSON)
+            .body(requestBody)
+            .when()
+            .post(url)
+            .then()
+            .statusCode(200)
+            .contentType(ContentType.JSON)
+            .body("id", notNullValue())
+            .body("resourceUri", containsString("/task/"))
+            .extract()
+            .path("id");
+
+    retryUntilTrue(
+        () -> {
+          List<Object> taskHistory =
+              get(getTestUrl("/task/" + taskId))
+                  .then()
+                  .contentType(ContentType.JSON)
+                  .extract()
+                  .path("history");
+          if (taskHistory
+              .toString()
+              .contains(String.format("Done creating 1 of %s", expectedServerGroupName))) {
+            return true;
+          }
+          return false;
+        },
+        String.format("Failed to detect service creation in %s seconds", TASK_RETRY_SECONDS),
+        TASK_RETRY_SECONDS);
+
+    ArgumentCaptor<RegisterTaskDefinitionRequest> registerTaskDefArgs =
+        ArgumentCaptor.forClass(RegisterTaskDefinitionRequest.class);
+    verify(mockECS).registerTaskDefinition(registerTaskDefArgs.capture());
+    RegisterTaskDefinitionRequest seenTaskDefRequest = registerTaskDefArgs.getValue();
+    assertEquals(expectedServerGroupName, seenTaskDefRequest.getFamily() + "-v000");
+    assertEquals(2, seenTaskDefRequest.getContainerDefinitions().size());
+    assertEquals(
+        80,
+        seenTaskDefRequest
+            .getContainerDefinitions()
+            .get(0)
```

Comment:
I'm not sure we can rely on the ordering of these to be the same every time. instead of getting the container def by index, can we find the container in the list by name and assert on all its props then?

Reply:
let's also assert on the name of each container (in addition to port)

Reply:
@allisaurus, AFAIK we can get container definitions from getContainerDefinitions() in the form of list. I tried to find the container in the list by name but that's not possible because there is no function available to find the container name from the list of container objects.
Do you think there is any other way by which we can figure that out?

let's also assert on the name of each container (in addition to port) -> I'm already asserting on container name on line 455-456.

Reply:
> that's not possible because there is no function available to find the container name from the list of container objects

you're right there's no built in function, but we can use an old fashion loop OR a fancy [filter expression w/ a lambda](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperation.java#L365-L368).

> I'm already asserting on container name on line 455-456

true! i'm just asking we moving it up so that all assertions on a given container are happening in the same place.